### PR TITLE
jtsync: Handle matrix jobs correctly

### PR DESCRIPTION
**Pull request by felixsch**
Add new logic for matrix jobs:
  Instead of setting the card on each status change the card is only set
  if a new build started or the build changes from successful to
  failed.

### CODE REVIEW

File: scripts/jtsync/jtsync.rb [hunk 1]
```diff
@@ -79,6 +85,10 @@ def card_name
     def list_name
       "OpenStack"
     end
+
+    def build_nr
```
Comment (rsalevsky) [inline scripts/jtsync/jtsync.rb]: The method is defined twice see line 74.

Comment (felixsch) [inline scripts/jtsync/jtsync.rb]: Yeah both SuseMatrix and OpenSuseMatrix need the build_nr method

Comment (rsalevsky) [inline scripts/jtsync/jtsync.rb]: Duplicated code is never good. ;)

**Comment by jdsn** [comment]
@felixsch 
Do these changes take into account that a Matrix Reloaded (partial rebuild) may succeed (because eg. only one configuration of the matrix is built) but not all configurations are green?

**Comment by felixsch** [comment]
Mhm. good question. No I would say no currently it's not aware of different builds for one card in general. You will get the status of the last build, no matter what configuration was used.
Maybe only trigger the script if it's not a manually triggered run?

**Comment by jdsn** [comment]
> Maybe only trigger the script if it's not a manually triggered run?

@felixsch 
Thats a nice idea, but it will hide the fixed and failed runs that are done via Matrix Rebuild. You also want to see them.
So the correct way would be to filter the configurations that are part of a specific run (a run/build) does not necessarily rebuild all valid configurations. So only update these in Trello that were affected (rebuilt).

**Comment by felixsch** [comment]
So the simplest partial solution is to disable all "jtsync" calls when a build in manually triggered. I will open a follow up pull request changing the calling behavior.

**Pull request merged**